### PR TITLE
Fix pool symbol and pool name disappearing

### PR DESCRIPTION
**Pull request by 1saf**
# Description

Was seeing the 'Missing pool name or prices' tooltip pop-up where it shouldn't have and after some deeper investigation, found the root cause for the problem. This hotfix addresses that by making sure that the pool name and symbol (autogenerated) are saved BEFORE loading the final step as well as make sure that the pool name and symbol are saved on click of the 'save' button rather than relying just on the update of the model value which seemed to be hit and miss.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

## How should this be tested?

Create a pool as normal, make sure to get to the final step. 
1. On the final step, refresh the page. It should keep the pool name and symbol as to what it was previously (autogenerated one)
2. Adjust the pool name and symbol yourself, then refresh the page. It should keep those.
3. On the final step, empty out the value for either pool name or symbol or both and make sure that an alert shows up and also disables the create pool button.

## Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code where relevant, particularly in hard-to-understand areas
- [X] My changes generate no new console warnings
- [X] The base of this PR is `master` if hotfix, `develop` if not

### Walkthrough

(Note, you will notice in the video that on refresh of the page, the symbols for the tokens dissapear! This was an injection bug which has since been fixed)
https://www.loom.com/share/b64e4de4606c4238895910897623d2fe

### CODE REVIEW

File: src/composables/pools/usePoolCreation.ts [hunk 1]
```diff
@@ -92,8 +92,12 @@ export default function usePoolCreation() {
         w => w.tokenAddress
       );
 
-      poolCreationState.name = getPoolSymbol();
-      poolCreationState.symbol = getPoolSymbol();
```
Comment (1saf) [inline src/composables/pools/usePoolCreation.ts:96]: This overrides the custom name or symbol that the user might have entered

**Review by timjrobinson — APPROVED** [review]
LGTM

**Review by EndymionJkb — APPROVED** [review]
Added the new key translations

File: src/composables/pools/usePoolCreation.ts [hunk 2]
```diff
@@ -92,8 +92,12 @@ export default function usePoolCreation() {
...
+      if (!poolCreationState.name) {
+        poolCreationState.name = getPoolSymbol();
+      }
```
Comment (garethfuller) [inline src/composables/pools/usePoolCreation.ts:97]: Is it possible to clean this up a bit with?
```suggestion
      !poolCreationState.name && poolCreationState.name = getPoolSymbol();
```
Same for conditional below if it works.

**Review by garethfuller — APPROVED** [review]
Looks good! Again just a couple of minor comments.

File: src/composables/pools/usePoolCreation.ts (hunk 2 above)
Comment (timjrobinson) [reply src/composables/pools/usePoolCreation.ts:97]: Could also be `poolCreationState.name = poolCreationState.name || getPoolSymbol()`

File: src/components/_global/BalInlineInput/BalInlineInput.vue [hunk 3]
```diff
@@ -98,6 +99,11 @@ const inputAttrs = computed(() => omit(attrs, 'class'));
  */
 function toggleEditable() {
   isEditable.value = !isEditable.value;
+  // if the value was changed to false it means that
+  // the user clicked the save icon
+  if (isEditable.value === false) {
+    emit('save');
+  }
```
[the remark this answers is not in the document]
Comment (1saf) [reply src/components/_global/BalInlineInput/BalInlineInput.vue:106]: done

**Pull request merged**